File: gc-app/src/main/java/rocks/frick/gc/application/ListActivityService.java
package rocks.frick.gc.application;

import java.util.List;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import rocks.frick.gc.Pageable;
import rocks.frick.gc.application.dto.FileIDTo;
import rocks.frick.gc.application.spi.ActivityRepository;

public class ListActivityService {

  private static final Logger log = LoggerFactory.getLogger(ListActivityService.class);

  private final ActivityRepository activityRepository;

  public ListActivityService(ActivityRepository activityRepository) {
    this.activityRepository = activityRepository;
  }

  public List<FileIDTo> listWorkouts(Pageable page) {
    log.debug("Listing all workouts");
    return activityRepository.findAll(page);
  }
}

File: gc-core/src/main/scala/gc/application/ListActivityService.scala
package gc.application

import gc.spi.FindAllActivities
import com.typesafe.scalalogging.Logger
import gc.Pageable
import gc.model.Activity

class ListActivityService(repo: FindAllActivities) {
  private val log = Logger("FindAllActivities")

  def listActivities(page: Pageable): List[Activity] = {
    log.debug(s"[CORE] Listing activities with page: $page.page, size: $page.size")
    repo.findAll(page)
  }

}
